task/07-yield-tasks.js

```diff
@@ -145,7 +203,14 @@ function* mergeSortedSequences(source1, source2) {
  *   Most popular implementation of the logic in npm https://www.npmjs.com/package/co
  */
 function async(generator) {
-    throw new Error('Not implemented');
+    let sourceGenerator = generator()
+
+    let managePromises = (result) => {
+        return result.done ?  Promise.resolve(result.value) : Promise.resolve(result.value).then((result) =>
```

Comment:
Please, avoid this code style, if-else is more readable than multiple conditional operators ? :